packages/dao-contracts/src/dao-contract.ts

```diff
@@ -0,0 +1,95 @@
+import { Token } from '@bitcoin-computer/TBC20'
+import { Contract } from '@bitcoin-computer/lib'
+
+type ElectionType = {
+  proposalMod: string
+  tokenRoot: string
+  description: string
+}
+
+type VoteType = {
+  electionId: string
+  tokens: Token[]
+  vote: 'accept' | 'reject'
+}
+
+export class Election extends Contract {
+  proposalMod!: string
+  tokenRoot!: string
+  description!: string
+  constructor({ proposalMod, tokenRoot, description }: ElectionType) {
+    super({ proposalMod, tokenRoot, description })
+  }
+
+  async proposalVotes(): Promise<string[]> {
+    const revs = await computer.getTXOs({ mod: this.proposalMod })
+    const voteTxIdsSet = new Set<string>(revs.map((r) => r.split(':')[0]))
+    const validVotes = new Set<string>(voteTxIdsSet)
+
+    for (const voteTxId of voteTxIdsSet) {
+      const ancestors = (await computer.getAncestors(voteTxId)) as string[]
+      const ancestorsSet = new Set<string>(ancestors)
+      ancestorsSet.delete(voteTxId)
+
+      if ([...ancestorsSet].some(voteTxIdsSet.has, voteTxIdsSet)) {
+        validVotes.delete(voteTxId)
+      }
+    }
+
+    return Array.from(validVotes)
+  }
+
+  private async validVotes(): Promise<Vote[]> {
+    const proposalVotes = await this.proposalVotes()
```

Comment:
Unless it is important for testing, I would suggest to inline `proposalVotes()` here because it is not so clear from the name what the function does. Otherwise we should try to find a better name.